source/extensions/tracers/common/ot/opentracing_driver_impl.cc

```diff
@@ -66,34 +66,18 @@ class OpenTracingHeadersReader : public opentracing::HTTPHeadersReader {
   }
 
   opentracing::expected<void> ForeachKey(OpenTracingCb f) const override {
-    // TODO(wbpcode): TraceContext currently does not provide an API to traverse all entries. So
-    // dynamic_cast has to be used here. This is a temporary compromise to ensure that the existing
-    // functions are correct. After TraceContext provides the iterative API, this part of the code
-    // needs to be rewritten.
-    const auto headers = dynamic_cast<const Http::RequestHeaderMap*>(&trace_context_);
-    if (headers != nullptr) {
-      headers->iterate(headerMapCallback(f));
-    }
+    trace_context_.iterateContext(
+        [cb = std::move(f)](absl::string_view key, absl::string_view val) {
+          opentracing::string_view opentracing_key{key.data(), key.length()};
+          opentracing::string_view opentracing_val{val.data(), val.length()};
+          cb(opentracing_key, opentracing_val);
+          return true;
```

Comment:
Is it ok that the callback's return value is ignored?

Reply:
@rojkov It not ok. Thanks for call it out.